src/YesSql.Core/Provider/IDbProviderOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using YesSql.Core.Services;
+
+namespace YesSql.Core.Provider
+{
+    public interface IDbProviderOptions
+    {
+        string ProviderName { get; set; }
```

Comment:
You don't seem to understand, so let me give you a more detailed explanation.

When we start Orchard CMS, we display a list of a supported database names: 
1- Sqlite, Sql Server, Postgres, MySql. 
2- The user selects one from this list.
3- We then show or hide the Connection String text input, and the Table Prefix one.
4- Once the user has entered these information we save it in configuration. 
5- The application is then reloaded and we can read the configuration to call UserSqlServer(cstr) or UseSqlite(cstr) ...based on what the user chose.

How do you think having a provider name in this class will help us list all the available providers on the step 1? I can tell you there is none. At that point we can't call this option, because we don't know what the user wants. Also where do you specify that a provider accepts connection strings or not? Again, not in this class. What exists already in Orchard CMS is both necessary and sufficient for listing the providers.

Now with your other changes, we can call UseXXX based on the configuration Orchard CMS will store.